js/stores/helpers/boundary-utils.ts

```diff
@@ -114,12 +114,22 @@ export function getBoundaryInfo(field: FieldStore, otherField: FieldStore): IBou
     }
   } else {
     // longitudinal boundary
-    const platePos = toSpherical(field.plate.center);
-    const otherPlatePos = toSpherical(otherField.plate.center);
-    // Order fields from west to east.
-    fields = platePos.lon < otherPlatePos.lon
-              ? [otherField, field]
-              : [field, otherField];
+    const { lat, lon } = toSpherical(field.absolutePos);
+    const fieldRotationAxis = field.absolutePos.clone().normalize();
+    const westTestVec = getNorthVector(lat, lon).setLength(0.5).applyAxisAngle(fieldRotationAxis, 0.5 * Math.PI);
+    const eastTestVec = getNorthVector(lat, lon).setLength(0.5).applyAxisAngle(fieldRotationAxis, -0.5 * Math.PI);
+    const westTestField = model.topFieldAt(field.absolutePos.add(westTestVec));
+    const eastTestField = model.topFieldAt(field.absolutePos.add(eastTestVec));
```

Comment:
`.add` adds the other vector in place. In this case, it modified `field.absolutePos`. AFAIR it's not a problem, as absolutePos is a getter that returns a new vector each time. But I've been trying to always .clone() it first, just to be safe. For example, if we decide to cache .absolutePos one day (probably would make sense given how many times it's requested).